### ISSUE TITLE
🐛 False positive of `useExportType` for `export {}`

### ISSUE THREAD

**Issue by alex-kinokon** [biomejs/biome#3535]
### Environment information

```block
CLI:
  Version:                      1.8.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v22.5.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.8.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

https://biomejs.dev/playground/?code=ZQB4AHAAbwByAHQAIAB7AH0A

### Expected result

No lint error.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by yossydev** [comment]
This was not mentioned in the [documentation](https://biomejs.dev/linter/rules/use-export-type/ ), but looking at the tests, this feels like a specification.

https://github.com/biomejs/biome/blob/9aea758a0173c4115b2e4415d77705ba383cf8ab/crates/biome_js_analyze/tests/specs/style/useExportType/invalid.ts#L31

**Comment by alex-kinokon** [comment]
I read the original PR and the one in `rome/tools`. Neither of it mentions this case in the description so I’m assuming this is not something intentional.

**Comment by Conaclos** [comment]
Is there an issue of turning `export` into `export type`? The intended behavior of turning a file into a module is still preserved?

**Comment by alex-kinokon** [comment]
The intention of `exportKind: "type"` is to ask the bundler to drop an export, and in this case there’s nothing to drop so the fix serves no purpose except leaving the reader wondering if this has some deliberate semantic difference.

**Comment by alex-kinokon** [comment]
Thanks!
